src/monodroid/jni/osbridge.cc

```diff
@@ -1041,7 +1041,7 @@ OSBridge::ensure_jnienv (void)
 	JNIEnv *env;
 	jvm->GetEnv ((void**)&env, JNI_VERSION_1_6);
 	if (env == nullptr) {
-		mono_thread_attach (mono_domain_get ());
+		mono_thread_attach (utils.get_current_domain (/* attach_thread_if_needed */ false));
```

Comment:
If `utils.get_current_domain()` calls `mono_thread_attach()` if the domain isn't set -- which presumably it *won't* be if this thread hasn't been seen before (?) --  then do we need this separate `mono_thread_attach()` call?

Why not just call `utils.get_current_domain()` and let the side-effect attach?

Reply:
We can't do that. If `mono_domain_get` does **not** return `NULL` then we wouldn't call `mono_thread_attach` at all. And it's most likely that `mono_domain_get` will **not** return `NULL`.